db/upgrade.php

```diff
@@ -650,5 +650,18 @@ function xmldb_attendance_upgrade($oldversion=0) {
         upgrade_mod_savepoint(true, 2021050700, 'attendance');
     }
 
+    if ($oldversion < 2021060700) {
+
+        // Changing precision of field statusset on table attendance_log to (1333).
+        $table = new xmldb_table('attendance_sessions');
+        $field = new xmldb_field('automarkcmid', XMLDB_TYPE_CHAR, '10', null, null, null, null, 'rotateqrcodesecret');
+
+        // Launch change of precision for field statusset.
+        $dbman->change_field_precision($table, $field);
```

Comment:
change_field_precision looks like the wrong upgrade api for adding a new field - this should be an add_field not a         
```
if (!$dbman->field_exists($table, $field)) {
     $dbman->add_field($table, $field);
{